controllers/metal3.io/baremetalhost_controller_test.go

```diff
@@ -1035,6 +1035,45 @@ func TestNeedsProvisioning(t *testing.T) {
 	}
 }
 
+// TestNeedsProvisioning verifies the logic for deciding when a host
+// needs to be provisioned when custom deploy is used.
+func TestNeedsProvisioningCustomDeploy(t *testing.T) {
```

Comment:
I'd find easier to understand this test if defined in the table driven mode, given the presence of multiple cases

Reply:
I'm not sure what you mean by "table driven mode", could you elaborate?

Reply:
Yes, sure, here's an example: https://github.com/metal3-io/baremetal-operator/blob/master/pkg/provisioner/ironic/updatehardwarestate_test.go.

Essentially in the test we're going to define a number of different relevant cases (each one with its own inputs and expectation), and such approach helps in reducing the duplications, reduce the risks of side effects, and definitely supports troubleshooting and maintenance.

(there's a great blog about it here https://dave.cheney.net/2019/05/07/prefer-table-driven-tests)